src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala

```diff
@@ -205,6 +204,61 @@ class UploadSpec extends DepositServletFixture {
     }
   }
 
+ it should "extract all files from a ZIP, with a nested zip" in {
+    File("src/test/resources/manual-test/nested.zip").copyTo(testDir / "input" / "2.zip")
+    val uuid = createDeposit
+    val relativeTarget = "path/to/dir"
+    val bagDir = testDir / "drafts" / "foo" / uuid.toString / "bag"
+    val absoluteTarget = (bagDir / "data" / relativeTarget).createDirectories()
+    absoluteTarget.list.size shouldBe 0 // precondition
+    post(
+      uri = s"/deposit/$uuid/file/$relativeTarget",
+      params = Iterable(),
+      headers = Seq(fooBarBasicAuthHeader),
+      files = Seq(("formFieldName", (testDir / "input/2.zip").toJava))
```

Comment:
just for readability:

```suggestion
      files = Seq("formFieldName" -> (testDir / "input/2.zip").toJava)
```